### PR TITLE
cover case where Poco is found, but the component is not

### PR DESCRIPTION
**Pull request by wjwwood**
This case occurs when a release version of poco is installed, but the workspace is built as debug, therefore it looks for (e.g.) `PocoFoundationd` but only `PocoFoundation` is available.

This resulted in poco being “found” but linking failing because `Poco_LIBRARIES` would not contain the Foundation library required.

I discovered this on macOS, but the build farm doesn’t notice because poco is explicitly not installed on those machines, but users might run into this problem where the build farm does not.

The result of this patch is that even when poco is found in release, when building a debug workspace poco will be downloaded and built as debug locally, and that local build will be used instead.

CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4034/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1125/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3363/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4126/)

### CODE REVIEW

File: cmake/Modules/FindPoco.cmake [hunk 1]
```diff
@@ -195,6 +195,10 @@ if(Poco_INCLUDE_DIR)
         if(LIB${COMPONENT})
           list(APPEND Poco_LIBRARIES "${LIB${COMPONENT}}")
           list(APPEND Comp_List "${COMPONENT}${DBG}")
+        else()
+          message(STATUS "  Poco found, but missing component 'Poco${COMPONENT}${DBG}'...")
+          message(STATUS "  This can happen when poco is found in release, but debug is required.")
```
Comment (dirk-thomas) [inline cmake/Modules/FindPoco.cmake:200]: The message describe one case (found release, requiring debug) but can also be triggered for the opposite case. Either the message should cover both cases or a different message should be shown based on `DBG`.

**Comment by wjwwood** [comment]
New CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4037/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1130/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3364/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4127/)

File: cmake/Modules/FindPoco.cmake (hunk 1 above)
Comment (dirk-thomas) [reply cmake/Modules/FindPoco.cmake:200]: The newly added second message is the same as the first one?

Comment (wjwwood) [reply cmake/Modules/FindPoco.cmake:200]: Doh, fixed it.

**Review by dirk-thomas — APPROVED** [review]

**Pull request merged**